### PR TITLE
Bump gl-native to v10.0.0-beta.22 and common to v12.0.0

### PR DESCRIPTION
**Pull request by kiryldz**
PRs must be submitted under the terms of our Contributor License Agreement [CLA](https://github.com/mapbox/mapbox-maps-android/blob/main/CONTRIBUTING.md#contributor-license-agreement).
Fixes: < Link to related issues that will be fixed by this pull request, if they exist >

## Pull request checklist:
 - [x] Briefly describe the changes in this PR.
 - [ ] Include before/after visuals or gifs if this PR includes visual changes.
 - [x] Write tests for all new functionality. If tests were not written, please explain why.
 - [ ] Add example if relevant.
 - [x] Document any changes to public APIs.
 - [x] Apply changelog label ('breaking change', 'bug :beetle:', 'build', 'docs', 'feature :green_apple:', 'performance :zap:', 'testing :100:') or use the label 'skip changelog'
 - [x] Add an entry inside this element for inclusion in the `mapbox-maps-android` changelog: `<changelog>Bump gl-native to v10.0.0-beta.22 and common to v12.0.0.</changelog>`.

### Summary of changes

Bump gl-native to v10.0.0-beta.22 and common to v12.0.0.

### Breaking changes

 - Rename `NetworkConnectivity` to `OfflineSwitch`
 - Remove `TileLoadOptions` from `TileRegionLoadOptions`. `networkRestriction` and `acceptExpired` can now be specified directly in `TileRegionLoadOptions`.
 -  Add `totalBytes` and `transferredBytes` to TileStoreLoadResult
 - `MapboxMap.setBounds` return type changed from Void to Expected
 - Expose tileset version for sources that use TileJSON metadata
 - ResourceRequest `offline-data` boolean field is replaced with the `source` string field, which whether the response came from network, cache or tile store.
 - Remove `Style.getStyleGeoJSONSourceClusterLeaves`, `Style.getStyleGeoJSONSourceClusterExpansionZoom`, `Style.getStyleGeoJSONSourceClusterChildren`. All those can be fully replaced by `MapboxMap.queryFeatureExtensions`.

### CODE REVIEW

File: LICENSE.md [hunk 1]
```diff
@@ -28,6 +28,30 @@ License: [The Apache Software License, Version 2.0](http://www.apache.org/licens
...
+===========================================================================
+
+Mapbox Maps Android uses portions of the Android Lifecycle LiveData Core.
+
+URL: [https://developer.android.com/topic/libraries/architecture/index.html](https://developer.android.com/topic/libraries/architecture/index.html)
+
+License: [The Apache Software License, Version 2.0](http://www.apache.org/licenses/LICENSE-2.0.txt)
+
+===========================================================================
+
```
Comment (tobrun) [inline LICENSE.md:54]: interesting additions, have we tracked down where these are coming from?

**Review by tobrun — COMMENTED** [review]
Should we document the breaking changes in the changelog or will they come from upstream? (eg. expected refactor, removal of stylemanager api's etc).

File: LICENSE.md (hunk 1 above)
Comment (kiryldz) [reply LICENSE.md:54]: No, I did not yet have time for that, just updated license by script. Do you have any ideas already?

File: LICENSE.md [hunk 2]
```diff
@@ -68,6 +108,126 @@ License: [The Apache Software License, Version 2.0](http://www.apache.org/licens
 
 ===========================================================================
 
+Mapbox Maps Android uses portions of the Android Support Library Coordinator Layout.
+
+URL: [http://developer.android.com/tools/extras/support-library.html](http://developer.android.com/tools/extras/support-library.html)
+
+License: [The Apache Software License, Version 2.0](http://www.apache.org/licenses/LICENSE-2.0.txt)
+
+===========================================================================
```
[the remark this answers is not in the document]
Comment (kiryldz) [reply LICENSE.md:117]: > Should we document the breaking changes in the changelog or will they come from upstream? (eg. expected refactor, removal of stylemanager api's etc).

Yes, when this PR will be moved from draft to ready for review - everything will be updated 👍

**Review by tobrun — APPROVED** [review]

**Comment by kiryldz** [comment]
@mapbox/maps-android FYI `build/start-run-style-test` was also run but before last commit, so we're safe to merge I believe.

**Pull request merged**